### ISSUE TITLE
pyproject.toml requires a [mypy] section?

### ISSUE THREAD

**Issue by alne4294** [python/mypy#14376]
**Documentation**

Hi there!  I'm new to mypy, but after a lot of digging in docs/google/github, I'm not seeing how `pyproject.toml` works... Ultimately, I had to switch to `my.ini` to get things to work.

**TLDR;**  I have the latest mypy version (v0.991) correctly pointing to the `pyproject.toml` configs, but the console output is complaining about multiple issues.  It seems to be parsing the `pyproject.toml` as a different config type. 

I tried downgrading to v0.960 without any luck.   I'm not sure if I'm doing something wrong, which is why I'm flagging it as a documentation issue.  

I called mypy via:
```
poetry run mypy ./myModule --config-file pyproject.toml
```

### Issue 1
```
pyproject.toml: No [mypy] section in config file
```

The section _was_ parsed when I just set it to `[mypy]`, but the [mypy docs](https://mypy.readthedocs.io/en/stable/config_file.html#using-a-pyproject-toml-file) clearly state that the section should be `[tool.mypy]` instead of `[mypy]`, which matches behavior of other commonly-used packages like `[tool.isort]`, `[tool.black]`, and `[tool.pylint.main]`.   

Only online reference I could find to this issue was [here](https://www.reddit.com/r/Python/comments/c5jn39/til_you_can_specify_your_mypy_pytest/) from 4 years ago on Reddit... 

### Issue 2
I was surprised that that mypy doesn't seem to be accepting double quotes for the python_version, when the docs say [here](https://mypy.readthedocs.io/en/stable/config_file.html#platform-configuration) it is a string: 
```
pyproject.toml: [mypy]: python_version: Invalid python version ''3.8'' (expected format: 'x.y')
```

It worked when I tossed the quotes and made it numeric.

### Issue 3
mypy wasn't honoring the mypy_path property.  I suspect this is because it had quotes based on past threads... but naturally the toml syntax will break without them.

Thanks for the time!

**Comment by hauntsaninja** [comment]
Something here seems really fishy. These errors don't really make any sense to me. Tbh it kind of sounds like your toml file is being parsed as an ini.

I can't reproduce any of this, would you be able to make a standalone repro?

**Comment by alne4294** [comment]
Yeah I agree... if you can't repro, let me look more closely to see if I can figure out why it's doing that...

**Comment by alne4294** [comment]
Okay I'm up and running with a pyproject.toml 🎉 ... Definitely an issue on my end, but I think it may relate to the caching for mypy and/or _possibly_ even poetry.  I cleared all mypy caches (including older toml configs/caches that I had renamed with an "old-" prefix since they seemed to be getting picked up) AND removed/re-added mypy thru poetry...

I was bumping from a busted setup + version that didn't support pyproject.toml (v0.812), so maybe that's partly what got sticky.

Thanks @hauntsaninja for the sanity check.

**Comment by hauntsaninja** [comment]
Awesome! Glad you figured it out. Yeah, if poetry used an old version of mypy that would do it

**Comment by alne4294** [comment]
yeah just tried to repro by re-running on the old version and then bumping, etc.... but couldn't.... oh well!  Thanks again.
